### PR TITLE
[WIP - Do not merge] Build support for OneCore target

### PR DESCRIPTION
**Pull request by gkhanna79**
1) Update the build system to locate and link against OneCore and uCRT (static or dynamic) Libraries
2) SOS still links against Desktop imports - will be filing issue to get this resolved independent of this change.
3) Update DelayLoadHelper mechanism to return last error, lookup API in an APISet binary on Win8+ and decouple module declaration from API declaration to allow a module to be declared once for multiple APIs exported from it.
4) coreclr.dll, mscordbi.dll and mscordaccore.dll links against OneCore imports for both Winx86 and Winx64. Arm64 is not yet supported.

@jkotas PTAL. You can ignore the TODOs since I need to fix them. I also need to fix the test run to pickup the APIsets.

@weshaggard @AlexGhiondea I believe we will need to remap mscorlib pinvokes. How do you propose this be done?

### CODE REVIEW

File: src/utilcode/longfilepathwrappers.cpp [hunk 1]
```diff
@@ -409,13 +420,25 @@ MoveFileWrapper(
...
         {
-            ret = MoveFileW(
-                    Existingpath.GetUnicode(),
-                    Newpath.GetUnicode()
-                    );
+            decltype(MoveFileW) *fpMoveFileW = nullptr;
+#if !defined(FEATURE_PAL)            
+            LOOKUP_API_VIA_APISET_FALLBACK(kernel32, api_ms_win_core_kernel32_legacy_l1_1_1, MoveFileW, &fpMoveFileW, lastError)
+#else // defined(FEATURE_PAL)            
+            fpMoveFileW = MoveFile;
```
Comment (jkotas) [inline src/utilcode/longfilepathwrappers.cpp]: We should call MoveFileEx instead of this.

**Comment by jkotas** [comment]
LGTM modulo comments and TODOs

**Comment by gkhanna79** [comment]
@jkotas The function in question is a wrapper for MoveFile - there is a seperate wrapper for MoveFileEx.

**Comment by gkhanna79** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

**Comment by gkhanna79** [comment]
Fixes https://github.com/dotnet/coreclr/issues/3014

**Comment by gkhanna79** [comment]
@dotnet-bot test  Ubuntu x64 Checked Build and Test please

**Pull request closed without merging**